### PR TITLE
GM: detect unplugged or missing radar

### PR DESCRIPTION
**Pull request by nworb-cire**
Force pushed to the branch so github won't let me reopen #26450

### CODE REVIEW

File: selfdrive/car/gm/values.py [hunk 1]
```diff
@@ -149,6 +149,13 @@ class CanBus:
...
+  },
+  {
+    170: 8, 171: 8, 189: 7, 190: 6, 192: 5, 193: 8, 197: 8, 199: 4, 201: 6, 209: 7, 211: 2, 241: 6, 288: 5, 289: 1, 290: 1, 298: 2, 304: 1, 308: 4, 309: 8, 311: 8, 313: 8, 320: 3, 328: 1, 352: 5, 368: 8, 381: 2, 384: 8, 386: 5, 388: 8, 389: 2, 390: 7, 417: 7, 419: 1, 426: 7, 451: 8, 452: 8, 453: 6, 454: 8, 456: 8, 458: 8, 479: 3, 481: 7, 485: 8, 489: 5, 493: 8, 495: 4, 497: 8, 499: 3, 500: 6, 501: 3, 508: 8, 512: 3, 528: 4, 530: 8, 532: 6, 537: 5, 539: 8, 542: 7, 546: 7, 550: 8, 554: 3, 558: 8, 560: 6, 562: 4, 563: 5, 564: 5, 565: 5, 566: 5, 567: 3, 568: 1, 573: 1, 608: 8, 609: 6, 610: 6, 611: 6, 612: 8, 613: 8, 647: 3, 707: 8, 711: 6, 761: 7, 810: 8, 821: 4, 823: 7, 832: 8, 840: 5, 842: 5, 844: 8, 853: 8, 866: 4, 961: 8, 967: 4, 969: 8, 977: 8, 979: 7, 988: 6, 989: 8, 995: 7, 1001: 5, 1003: 5, 1005: 6, 1009: 8, 1017: 8, 1019: 2, 1020: 8, 1033: 7, 1034: 7, 1105: 6, 1187: 4, 1217: 8, 1221: 5, 1223: 3, 1225: 7, 1227: 4, 1233: 8, 1249: 8, 1257: 6, 1265: 8, 1267: 1, 1273: 3, 1275: 3, 1280: 4, 1300: 8, 1322: 6, 1323: 4, 1328: 4, 1417: 8, 1905: 7, 1906: 7, 1907: 7, 1910: 7, 1912: 7, 1922: 7, 1927: 7, 2024: 8, 2025: 8, 2028: 8
```
Comment (sshane) [inline selfdrive/car/gm/values.py:158]: Why two dicts?

File: selfdrive/car/gm/interface.py [hunk 2]
```diff
@@ -112,6 +114,7 @@ def _get_params(ret, candidate, fingerprint, car_fw, experimental_long):
     ret.radarTimeStep = 0.0667  # GM radar runs at 15Hz instead of standard 20Hz
 
     if candidate == CAR.VOLT:
+      ret.minEnableSpeed = -1
```
Comment (sshane) [inline selfdrive/car/gm/interface.py:117]: Careful that some older Volts fault if you engage under a certain speed for the first time, I'd leave this. Or at the least, detect this somehow

Comment (nworb-cire) [reply selfdrive/car/gm/interface.py]: Thanks for the catch -- didn't mean to include that change in the commit.

File: selfdrive/car/gm/values.py (hunk 1 above)
Comment (nworb-cire) [reply selfdrive/car/gm/values.py:158]: If I recall correctly, one of these is mine, and the other belongs to a fellow flashed '18 Volt driver. I can double check exactly which is whose if you'd like.

Comment (sshane) [reply selfdrive/car/gm/values.py:158]: Looks like the second FP is missing the messages from the first one you added: `ASCM_2CD`, `AEBCmd`, `ASCM_365`, `ASCM_78A`.

One car has AEB and an ASCM and one doesn't? But neither have a radar?

File: selfdrive/car/gm/interface.py [hunk 3]
```diff
@@ -85,7 +86,7 @@ def _get_params(ret, candidate, fingerprint, car_fw, experimental_long):
     else:  # ASCM, OBD-II harness
       ret.openpilotLongitudinalControl = True
       ret.networkLocation = NetworkLocation.gateway
-      ret.radarUnavailable = False
+      ret.radarUnavailable = RADAR_HEADER_MSG not in fingerprint[CanBus.OBSTACLE] or DBC[candidate]["radar"] is None
```
Comment (sshane) [inline selfdrive/car/gm/interface.py:89]:
```suggestion
      ret.radarUnavailable = RADAR_HEADER_MSG not in fingerprint[CanBus.OBSTACLE]
```

**Comment by sshane** [comment]
Awaiting an updated CAN fingerprint and should be good to go! Any objections @adeebshihadeh?

**Pull request merged**